### PR TITLE
[Documentation] Cosmovisor 

### PR DESCRIPTION
**Pull request by okdas**
## Summary

Adds documentation on how to deploy a Validator and a Full Node using Cosmovisor. 

## Issue

- #526 

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [x] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Documentation**: `make docusaurus_start`; only needed if you make doc changes
- [ ] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [ ] I have commented my code
- [ ] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: docusaurus/docs/operate/run_a_node/full_node_cosmovisor.md
Comment (okdas) [inline docusaurus/docs/operate/run_a_node/full_node_cosmovisor.md:39]: I think having a script would be easier than listing commands one-by-one.

**Comment by okdas** [comment]
Note for the reviewer:
- The URL of the script is not live until this is merged. Use `https://raw.githubusercontent.com/pokt-network/poktroll/dk-docs/tools/installer/full-node.sh` if you want to pull the script from this branch;
- Similarly, the config changes are not live until the PR is merged, and the release is cut. In order for the node to start successfully, the `min-gas-prices` needs to be adjusted in `app.toml`. With this code, that will become the default value when we provision the configs.

File: tools/installer/full-node.sh
Comment (okdas) [reply tools/installer/full-node.sh:181]: you trippin

**Comment by okdas** [comment]
Added cosmovisor to container images in #796.

File: config.yml
Comment (Olshansk) [inline config.yml:83]:
```suggestion
      # Ignite does not carry over all defaults, so we are going to match `minimum-gas-prices` with `cmd/config.go`.
      # See the enhancement request here: https://github.com/ignite/cli/issues/4340
```

File: config.yml
Comment (Olshansk) [inline config.yml:81]:
```suggestion
    # BUG_DEV_NOTE: To change the default configs (that can be created using `poktrolld init` - adjust `cmd/poktrolld/cmd/config.go`)
```

File: config.yml
Comment (Olshansk) [inline config.yml:83]:
```suggestion
      # BUG_DEV_NOTE: Ignite does not carry over all defaults, so we are going to match `minimum-gas-prices` with `cmd/config.go`.
```

File: cmd/poktrolld/cmd/config.go
Comment (Olshansk) [inline cmd/poktrolld/cmd/config.go:89]:
```suggestion
	// Set default values so `poktrolld init` creates configs with "blessed" values:
	// BUG_DEV_NOTE: See `config.yml` for details why these have to to be set manually
```

File: tools/installer/full-node.sh
Comment (Olshansk) [inline tools/installer/full-node.sh:1]: I'm not going to review this bash code, but I have a question and a request:

1. Request - Can you add a make target to trigger this?
2. Question - Should this work on my local machine if I just run it as is?

**Review by Olshansk — CHANGES_REQUESTED** [review]
Pushed some changes and leaving some comments.

Great job!

File: cmd/poktrolld/cmd/config.go
Comment (okdas) [reply cmd/poktrolld/cmd/config.go]: Changed the wording.

File: config.yml
Comment (okdas) [reply config.yml:83]: This is not a bug, l'll leave DEV_NOTE though.

File: cmd/poktrolld/cmd/config.go
Comment (okdas) [reply cmd/poktrolld/cmd/config.go]: `config.yml` only changes values for the first validator. We use this on LocalNet only.

File: tools/installer/full-node.sh
Comment (okdas) [reply tools/installer/full-node.sh:1]: This script is intended to be downloaded and ran on the machine (something like https://get.docker.com/). As a result:
1. When would we need a make target for it if we should only run the script on something like VPS/Dedicated Server?
2. Negative, the script is not meant to run in a non-linux environment, and I have not tested WSL2.

File: cmd/poktrolld/cmd/config.go
Comment (Olshansk) [inline cmd/poktrolld/cmd/config.go:41]:
```suggestion
// but it helps ensure most of them are using the same configuration; as a baseline.
```

File: cmd/poktrolld/cmd/config.go
Comment (Olshansk) [inline cmd/poktrolld/cmd/config.go:42]: I'm not completely sure what you mean by this sentence. Please update or delete.

**Review by Olshansk — APPROVED** [review]
A couple small comments but looks good otherwise.

File: cmd/poktrolld/cmd/config.go
Comment (okdas) [reply cmd/poktrolld/cmd/config.go]: Pushed the new version that I think better explains what's going on.

I'll merge it, but if you don't like it for some reason, please let me know and I'll address.

Thank you for your review!

**Pull request merged**
